pallets/orml-currencies-allowance-extension/src/lib.rs

```diff
@@ -298,7 +287,10 @@ impl<T: Config> Pallet<T> {
 				if remaining.is_zero() {
 					*maybe_approved = None;
 				} else {
-					approved = remaining;
+					//decrement allowance only if it isn't max value (which acts as infinite allowance)
+					if approved.clone().checked_into::<u128>().unwrap_or_default() != u128::MAX {
```

Comment:
The `amount` is of type `BalanceOf<T>`. This can be any number type (fulfilling some traits). Let's try not to explicitly use `u128` here. I'd rather try to use the maximum value of the `Balance` type in some way. If I saw correctly, we can use the `max_value()` function on the `Balance` type because it implements the `Bounded` trait.

Reply:
Done. Thanks!